packages/core/src/components/alert/alert.tsx

```diff
@@ -48,7 +48,7 @@ export interface IAlertProps extends IOverlayLifecycleProps, IProps {
     confirmButtonText?: string;
 
     /** Name of a Blueprint UI icon (or an icon element) to display on the left side. */
-    icon?: IconName | JSX.Element;
+    icon?: IconName | OptionalElement;
```

Comment:
`IconName | React.ReactNode` unfortunately resolves to just `React.ReactNode` because nodes allow `string` values so we lose all the safety of the enum.